### PR TITLE
feat: add support for ESP32-S3-WROOM-1-N16R8

### PR DESCRIPTION
**Pull request by BA7LYA**
Add support for ESP32-S3-WROOM-1-N16R8 module.

[datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-s3-wroom-1_wroom-1u_datasheet_en.pdf)

### CODE REVIEW

**Review by thijstriemstra — APPROVED** [review]
works for me.

**Comment by naschpitz** [comment]
Sorry about my ignorance, but why is the `maximum_ram_size` so large? Shoudn't it be just 512 KB? The other 8 MB aren't exatcly RAM, I guess, as it can't be used as stack memory, only as heap memory by calling `ps_malloc()`.

**Comment by 19e2k67** [comment]
Development version model: ESP32-S3-WROOM-1-N16R8, platformio.ini configuration is as follows:
```c
[env:esp32s3]
platform = espressif32
board = esp32-s3-devkitc-1
framework = arduino
; 指定为16MB的FLASH分区表
board_build.arduino.partitions = default_16MB.csv
; 指定FLASH和PSRAM的运行模式
board_build.arduino.memory_type = qio_opi
; 预定义宏，启用PSRAM
build_flags = -DBOARD_HAS_PSRAM
; 指定FLASH容量为16MB
board_upload.flash_size = 16MB
```
Reference link: [Reference link](https://www.cnblogs.com/macrored/p/17357581.html)

File: boards/esp32-s3-wroom-1-n16r8.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        "f_flash": "80000000L",
+        "flash_mode": "qio",
+        "hwids": [
+            [
+                "0x303A",
+                "0x1001"
+            ]
+        ],
+        "mcu": "esp32s3",
+        "variant": "esp32_s3_wroom_1_n16r8"
```
Comment (mkalkbrenner) [inline boards/esp32-s3-wroom-1-n16r8.json:27]: Using this board file in my own projects let `#include <Arduino.h>` lead to this error:
`framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:29:10: fatal error: pins_arduino.h: No such file or directory`.

If I replace the variant name `esp32_s3_wroom_1_n16r8` by `esp32s3`, my projects compile!

File: boards/esp32-s3-wroom-1-n16r8.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            "ldscript": "esp32s3_out.ld",
+            "memory_type": "qio_opi",
+            "partitions": "default_16MB.csv"
+        },
+        "core": "esp32",
+        "extra_flags": [
+            "-DBOARD_HAS_PSRAM",
+            "-DARDUINO_ESP32_S3_N16R8",
+            "-DARDUINO_USB_MODE=1",
+            "-DARDUINO_USB_CDC_ON_BOOT=1",
```
Comment (chrBrd) [inline boards/esp32-s3-wroom-1-n16r8.json:13]: `-DARDUINO_USB_CDC_ON_BOOT=1` probably shouldn't be enabled at this level, but in `platform.io` if it's needed.

I'm using this chip on an [ESP32-S3-DevKitC-1](https://docs.espressif.com/projects/esp-idf/en/stable/esp32s3/hw-reference/esp32s3/user-guide-devkitc-1.html). If I include this flag is included then I have to hold down the 'Boot' button and press 'Reset' _every time_ I want to reflash. If this flag is removed then I can just 'Upload and Monitor' as normal.

Comment (sivar2311) [reply boards/esp32-s3-wroom-1-n16r8.json:13]: > `-DARDUINO_USB_CDC_ON_BOOT=1` probably shouldn't be enabled at this level, but in platform.io if it's needed.

Same for `-DARDUINO_USB_MODE=1` !

File: boards/esp32-s3-wroom-1-n16r8.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            ]
+        ],
+        "mcu": "esp32s3",
+        "variant": "esp32_s3_wroom_1_n16r8"
+    },
+    "connectivity": [
+        "bluetooth",
+        "wifi"
+    ],
+    "debug": {
```
Comment (chrBrd) [inline boards/esp32-s3-wroom-1-n16r8.json:33]: This section should include:
```json
    "default_tool": "esp-builtin",
    "onboard_tools": [
      "esp-builtin"
    ],
```

to enable the built-in debugger.

File: boards/esp32-s3-wroom-1-n16r8.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        "arduino",
+        "espidf"
+    ],
+    "name": "ESP32-S3-WROOM-1-N16R8",
+    "upload": {
+        "flash_size": "16MB",
+        "maximum_ram_size": 327680,
+        "maximum_size": 16777216,
+        "require_upload_port": true,
+        "speed": 460800
```
Comment (chrBrd) [inline boards/esp32-s3-wroom-1-n16r8.json:46]: This can safely be set to 921600 bauds.

**Comment by trip5** [comment]
All these changes look good and I can get programs to compile and upload with all of the suggestions.  What's the hold up on getting it approved?  Is it because @BA7LYA hasn't been active since July?

File: boards/esp32-s3-wroom-1-n16r8.json (hunk 1 above)
Comment (coloboxp) [reply boards/esp32-s3-wroom-1-n16r8.json]: Same behaviour here, trying to compile for Firebeetle2 N16R8; not sure if by setting it as `esp32s3` we are "missing some cool optimization" or so

**Comment by sivar2311** [comment]
If this gets merged, what about all the other available ESP32-S3 configurations?

I would recommend using the default ESP32-S3 DevKit C and make the necessary modifications in `platformio.ini`.

See https://github.com/sivar2311/ESP32-S3-PlatformIO-Flash-and-PSRAM-configurations

**Comment by trip5** [comment]
Oh wow.  @Sivar2311 your documentation has improved even more since last I checked.  You've got them all!! ...and I only have the N16R8 but that still looks incredibly useful... This seems like a very good solution for everyone...!

**Pull request closed without merging**